### ISSUE TITLE
podman_image: using force is required to pull existing changed image from registry

### ISSUE THREAD

**Issue by cwchristerw** [containers/ansible-podman-collections#879]
There is currently no way to use pull similarly to "podman pull" command. Which will check if image is same and dont waste Docker Hub quota, as far I know. I can run podman pull multiple times without quota being wasted example command "watch -n 0.1 'podman pull docker.io/guacamole/guacamole'.

I think using pull, this module should check from registry if it has newer image or digest has changed, then pull image only when it has changed. I would like to not use force parameter, because it will waste limited quota very fast. In my own registry there is no quota and I can continue to use force: true, but in case of Docker Hub this isn't possible.

I will build my own versions of these images in single machine and then push them to my own Gitea instance.

This might be related to issue in the past #153.

**Comment by cwchristerw** [comment]
There is no module to get latest image information from repository, which causes this problem to be currently only solvable by using uri module and using HEAD request.

**Comment by sshnaidm** [comment]
Podman doesn't support such functionality, there is a [Skopeo](https://github.com/containers/skopeo/blob/main/docs/skopeo.1.md) tool for this. I don't think image module should require a Skopeo.
But I'm open to a separate Skopeo module.

**Comment by sshnaidm** [comment]
related to #876 as well

**Comment by cwchristerw** [comment]
If its not possible to make it with podman itself then skopeo module would be good solution to make.

**Comment by sshnaidm** [comment]
Yeah, I'd like to have it. Would you like to prepare a patch 😄 ?

**Comment by cwchristerw** [comment]
Unfortunately I have no time and original reason why I suggested this wasn't solution to my real problem with Docker Hub quota. Docker Hub requires clients to login when getting digest in HEAD request from images and I'm not aware of what scope should be to login once to get info about all public images. But in current documentation of Docker Hub API it will have scope of one image and this will consume quota as fast as issuing podman pull directly. I'm not sure if skopeo has solved this quota issue itself somehow, if they have then it could be solution, but currently I have too much work todo before the end of the year.

**Comment by sgflt** [comment]
I wanted to achieve similar results, but failed.

[pull](https://docs.ansible.com/ansible/latest/collections/containers/podman/podman_container_module.html#parameter-pull): newer | always
[recreate](https://docs.ansible.com/ansible/latest/collections/containers/podman/podman_container_module.html#parameter-recreate): false | true
_force: false | true_ - couldn't find this config in [documentation](https://docs.ansible.com/ansible/latest/collections/containers/podman/podman_container_module.html)


**recreate: true** - always recreates container even if image stays the same 🔴 
**pull: newer** - does nothing, remote image is newer and local is stale. Even `podman pull myimage:latest` on machine and then repeated run of ansible task does not recreate container with new image. 🔴 

`ansible-playbook [core 2.16.12]`
`python version = 3.13.0 (main, Oct  8 2024, 00:00:00) [GCC 14.2.1 20240912 (Red Hat 14.2.1-3)]`
`jinja version = 3.1.4`

**Comment by sshnaidm** [comment]
@sgflt in order to use the newest image that exists locally with same tag exactly you need to use `image_strict: true`:

**Comment by sshnaidm** [comment]
I've started to work on HEAD requests for dockerhub in `podman_image`, thinking about what option to use for this in `podman_container`

**Comment by sgflt** [comment]
I have tested your recommendation, but still it does not discover newer image in registry.

Even following combination does not redeploy changed image: 🔴 
```yaml
pull: always
image_strict: true
```
But if I pull manually and then run playbook, then only the updated image is redeployed. 🟠

**Comment by sshnaidm** [comment]
> I have tested your recommendation, but still it does not discover newer image in registry.

Yes, this is exactly what I meant. It will discover it locally, currently it doesn't communicate with registry.
